zproject/default_settings.py

```diff
@@ -1,33 +1,38 @@
+import os
 from typing import TYPE_CHECKING, Any, Dict, List, Optional
+from urllib.parse import SplitResult
+
+from .config import DEVELOPMENT, PRODUCTION, get_secret
 
 if TYPE_CHECKING:
     from django_auth_ldap.config import LDAPSearch
     from typing_extensions import TypedDict
 
-from .config import DEVELOPMENT, PRODUCTION, get_secret
-
 if PRODUCTION:
     from .prod_settings import EXTERNAL_HOST, ZULIP_ADMINISTRATOR
 else:
     from .dev_settings import EXTERNAL_HOST, ZULIP_ADMINISTRATOR
 
-import os
-
 DEBUG = DEVELOPMENT
 
+# The hostname part of EXTERNAL_HOST (no port number)
+r = SplitResult("", EXTERNAL_HOST, "", "", "")
+assert r.hostname is not None
+EXTERNAL_HOST_WITHOUT_PORT = "[" + r.hostname + "]" if ":" in r.hostname else r.hostname
```

Comment:
Should we make this logic a function in `zulip_tools.py`?  Seems like we call it from a few places.  And in that function, I think the `[]` logic needs a comment about IPv6; folks not used to IPv6 likely will think "what are these `[`s doing here?".

Reply:
The annoying detail is that the other copy of this logic is in `dev_settings` which can’t import `zulip_tools` because it needs work in Python 2 for `zthumbor`.

Reply:
Annoying; well we can have 2 copies with comments if that's simpler; I'd still put one in `zulip_tools.py` and the other can note that it's a duplicate required by Python 2 support (so we can know to fix it later when we finally upgrade or dump thumbor).

Reply:
Done.